### DISCUSSION TITLE
Can't run cfclient on ubuntu

### DISCUSSION THREAD

**Discussion by Abhinandan-Krishnan** [bitcraze/discussions#951] [Q&A]
Hello, I installed cfcleint from the source and followed all the steps as per the docs. When I run cfclient, I get the belwo error,

```
INFO:cfclient.gui:Disabling STL printouts
INFO:cfclient.utils.input.inputreaders:Input readers: ['linuxjsdev', 'pysdl2']
INFO:cfclient.utils.input.inputreaders:Successfully initialized [linuxjsdev]
INFO:cfclient.utils.input.inputreaders:Could not initialize [pysdl2]: No SDL2 support on Linux
INFO:cfclient.utils.input.inputinterfaces:Found interfaces: ['leapmotion', 'wiimote', 'zmqpull']
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [leapmotion]: Leap Motion library probably not installed (No module named 'leapsdk')
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [wiimote]: Missing cwiid (wiimote) driver No module named 'cwiid'
INFO:cfclient.utils.config:Dist config read from /home/quadrotarium/crazyflie-clients-python/src/cfclient/configs/config.json
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [zmqpull]: ZMQ input disabled in config file
INFO:cfclient.utils.zmq_led_driver:ZMQ led disabled in config file
INFO:cfclient.utils.zmq_param:ZMQ param disabled in config file
INFO:cfclient.gui:If startup fails because of "xcb", install dependency with `sudo apt install libxcb-xinerama0`.
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: minimal, eglfs, xcb, linuxfb, minimalegl, vnc, wayland, vkkhrdisplay, offscreen, wayland-egl.

Aborted (core dumped)
```

please let me know how to resolve this.

**Comment by knmcguire** [comment]
Hi! I've transferred your issue to a discussion and edited your question to have the code in the right format.

 Which operating system are you using? 

The error message suggests that you are missing a package  `sudo apt install libxcb-xinerama0`. This is one of the requirements to have the cfclient running on your machine. This is the documentation for Ubuntu:  https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/#debianubuntu

**Reply by Abhinandan-Krishnan** [reply]
Thank you for the response. I am using Ubuntu 22. 
I tried installing `sudo apt install libxcb-xinerama0`. The issue still persists. 

it says `Reinstalling the application may fix this problem.`. How do I delete the current installation completely?

**Reply by knmcguire** [reply]
This might be due to a conflict of another library that is installed... Have you tried installing the cfclient in a virtual environment? Like venv? https://docs.python.org/3/library/venv.html

Or have you tried installing the cfclient not from source but the latest release? This requires uninstall of the original source version. 

```
pip3 uninstall cfclib cfclient
pip3 install cfclient
```

**Reply by knmcguire** [reply]
hi! and, how did it go?

There seems to be some additional issues of QT6 that needed extra dependencies. I had to additionally install 

`
sudo apt-get install libxcb-cursor0
`

but my error message was a bit different than yours

**Reply by Abhinandan-Krishnan** [reply]
I still have issues installing it globally, but I created a conda env and installed it from the source inside the env. Works fine.

**Reply by krichardsson** [reply]
There is a list of all libs that Qt6 requires at https://doc.qt.io/qt-6/linux-requirements.html
